### PR TITLE
make it possible to reference a branch of a local git repository

### PR DESCRIPTION
**Pull request by vn971**
Reasoning on why this change is wanted can be found
on the "sbt-dev" mailing list,
topic [problem with referencing a local git repository](https://groups.google.com/forum/#!topic/sbt-dev/uWBMBOII3GM).

```
file:///home/vasya/dev/repo.git#branch
```

### CODE REVIEW

**Comment by jsuereth** [comment]
LGTM

File: util/io/src/main/scala/sbt/IO.scala [hunk 1]
```diff
@@ -832,7 +832,7 @@ object IO {
     {
       if (!uri.isAbsolute) return uri; //assertAbsolute(uri)
       val str = uri.toASCIIString
-      val dirStr = if (str.endsWith("/") || uri.getScheme != FileScheme) str else str + "/"
+      val dirStr = if (str.endsWith("/") || uri.getScheme != FileScheme || uri.getRawFragment != null) str else str + "/"
```
Comment (eed3si9n) [inline util/io/src/main/scala/sbt/IO.scala]: This is a personal preference, but I prefer `Option(uri.getRawFragment).isDefined` over `null` in the code.

Comment (vn971) [inline util/io/src/main/scala/sbt/IO.scala]: Should I change?

Comment (eed3si9n) [inline util/io/src/main/scala/sbt/IO.scala]: Yea, could you? Unless you have some good reason to keep `null` in there. Otherwise, I'm ready to merge.

**Comment by eed3si9n** [comment]
Merged. Thanks for the contribution.

**Pull request merged**